### PR TITLE
Race condition fixed!  I hope! :)

### PR DESCRIPTION
**Pull request by danwalkowski**
Checking for initSession being in progress seems to eliminate the race condition on startup.
@aaustin @derrickstaten

### CODE REVIEW

**Comment by danwalkowski** [comment]
@Sarkar

File: Branch-SDK/Branch-SDK/Branch.m [hunk 1]
```diff
@@ -1252,6 +1228,13 @@ - (void)processNextQueueItem {
 
 #pragma mark - Session Initialization
 
+// appears to fix race condition on startup, when initSession is still in progress - DAW
+- (void)initSessionIfNeededAndNotInProgress {
```
Comment (Sarkar) [inline Branch-SDK/Branch-SDK/Branch.m]: I love the descriptive ness!

**Comment by derrickstaten** [comment]
@Sarkar can you test this an make sure the initSession callback always fires and you don't see other strange behavior?

**Comment by Sarkar** [comment]
I did yesterday, and I didn't witness anything out of the ordinary, tried a
variety of scenarios (open link from app background, cold start, etc.)

On Tue, Nov 10, 2015 at 1:58 PM, derrickstaten <notifications@github.com>
wrote:

> @Sarkar <https://github.com/Sarkar> can you test this an make sure the
> initSession callback always fires and you don't see other strange behavior?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BranchMetrics/iOS-Deferred-Deep-Linking-SDK/pull/219#issuecomment-155580837>
> .
>



-- 
[image: Branch Metrics mobile deep linking] <http://branch.io> * Sahil
Verma*
 Solutions Engineer | Branch Metrics
 m 510-386-1750 | e sahil@branch.io

*What is deep linking? Find out now.
<https://t.xink.io/Tracking/Index/oAwAADcvAABlTAYA0> Enter the $10k Mobile
Growth Challenge - get details here.
<https://t.xink.io/Tracking/Index/nAwAADcvAABlTAYA0>*

**Comment by derrickstaten** [comment]
Awesome! I wanted to make sure I spread the blame/responsibility in case this leads to issues down the line. I'll thumb after I test a bit more.

**Comment by derrickstaten** [comment]
:+1:

**Pull request merged**